snapcraft/tests/integration/general/test_build_package_grammar.py

```diff
@@ -130,6 +130,16 @@ def test_to_other_arch(self):
         self.run_snapcraft(['pull'])
         self.assertFalse(self._hello_is_installed())
 
+    def test_on_to_other_arch(self):
+        """Test that 'on to' fetches nothing when building for another arch."""
+
+        with self.modified_yaml() as yaml:
+            yaml['parts']['my-part']['build-packages'] = [
+                OrderedDict({'on i386 to other-arch': ['hello']})
```

Comment:
I have similar misgivings on this PR's integration tests to the ones I have on #1639.